File: 프로그래머스/2/17686. ［3차］ 파일명 정렬/［3차］ 파일명 정렬.py
import re
def solution(files):
    answer = []
    dic = []
    
    for file in files:
        temp = re.split('([0-9]+)', file)
        dic.append((file, temp[0].lower(), int(temp[1])))
        
            
    print(dic)
    dic.sort(key=lambda x:x[2])
    dic.sort(key=lambda x:x[1])
    answer = [file[0] for file in dic]
    
    return answer

File: 프로그래머스/2/17686. ［3차］ 파일명 정렬/［3차］ 파일명 정렬.js
function solution(files) {
    var answer = [];
    const sortedFile = [];  // 파일명, HEAD, NUMBER
    
    for (let file of files){
        let f = [file];
        let splitFile = file.split(/(\d+)/);
        f.push(splitFile[0].toLowerCase());
        let number = splitFile[1].length>5?splitFile[1].slice(0,6):splitFile[1];
        f.push(Number(number));
        sortedFile.push(f);
    }
    
    sortedFile.sort((a, b) => {
        // 첫번째 문자 비교정렬
        if (a[1] < b[1]) return -1;
        if (a[1] > b[1]) return 1;
        // 같으면 숫자 비교정렬
        return a[2] - b[2];
    });
    
    answer = sortedFile.map((value) => value[0]);
    return answer;
}